Lib/test/test_asyncio/test_timeouts.py

```diff
@@ -0,0 +1,208 @@
+"""Tests for asyncio/timeouts.py"""
+
+import unittest
+import time
+
+import asyncio
+from asyncio import tasks
+
+
+def tearDownModule():
+    asyncio.set_event_loop_policy(None)
+
+
+class BaseTimeoutTests:
+    Task = None
+
+    def new_task(self, loop, coro, name='TestTask'):
+        return self.__class__.Task(coro, loop=loop, name=name)
+
+    def _setupAsyncioLoop(self):
+        assert self._asyncioTestLoop is None, 'asyncio test loop already initialized'
+        loop = asyncio.new_event_loop()
```

Comment:
Note to myself: extract a part of the method into a public overridable one, e.g. `def setupEventLoop(self) -> AbstractEventLoop`